### PR TITLE
Issue 95: i really like my DC to be generated please

### PR DESCRIPTION
**Pull request by DiegoPino**
My first humble attempt on actually triggering the MODS to DC when global DC transform is not enabled. It's still -humble- and I would like to generalize the code a bit more. This is only for ingests yet, not for updates. Also deals with the elusive & so missing, COLLECTION POLICY. 

I have not tested the code still, mostly blindly coded, but after a few nightly passes, it should work. Will keep on working on this tomorrow AM and final, merge-able version, should go into master soon after that. 

NOTE: I did not indent code that was not touched to avoid cluttering any code review, you (me expecting) could help me doing here.

### CODE REVIEW

**Comment by DiegoPino** [comment]
@patdunlavey could I ask you to test this? The procedure is simple.
1.- Disable  Default DC XSLT from Islandora XML forms settings form (save)
2.- Run any ingest (try with a few objects and a collection?) selecting default DC XLST for the DC datastream. I added an empty Collection policy creation in this pull too. 
Only if you have some time of course. Thanks a lot

**Comment by patdunlavey** [comment]
@DiegoPino I will test this today and report back. Thanks!

File: includes/islandora_multi_batch.inc [hunk 1]
```diff
@@ -481,52 +481,102 @@ class IslandoraMultiBatchObject extends IslandoraBatchObject {
...
-        $ds['file']->uri :
-        $ds['datastream_file'];
-
-      $datastream = $this->constructDatastream($ds['dsid'], $ds['control_group']);
-      $datastream->label = $ds['label'];
-      $datastream->mimetype = $ds['mimetype'];
+          $datastream->setContentFromFile($ds_uri);
+          $this->ingestDatastream($datastream);
+          $dowehaveMODS = $ds['dsid'] == "MODS" ? true : false; 
+          $dowehaveDC = $ds['dsid'] == "DC" ? true : false;
```
Comment (patdunlavey) [inline includes/islandora_multi_batch.inc]: There's an error in the logic here for tracking the presence of MODS and DC datastreams. Will be false if the DS in question is not the last one.
Should be something like:
          if($ds['dsid'] == "MODS") $dowehaveMODS = TRUE;
          if($ds['dsid'] == "DC") $dowehaveDC = TRUE;

Comment (DiegoPino) [reply includes/islandora_multi_batch.inc]: Totally, what a mess. fixing now

Comment (DiegoPino) [reply includes/islandora_multi_batch.inc]: So what about $dowehaveMODS = $ds['dsid'] == "MODS" ? true : $dowehaveMODS;  ?
I think that was my original idea.

Comment (DiegoPino) [reply includes/islandora_multi_batch.inc]: So found another error after fixing this one ( on the collection policy!), fixed too. The worst way of saying goodbye to this year, brain is mashed potatoes. New logic assigns the previous value, so if set to true at any point, true will stick. Can you check my brain is doing the right logic there? Thanks

Comment (patdunlavey) [reply includes/islandora_multi_batch.inc]: Works for me. Seems slightly redundant to be assigning on each time through the loop when we know that it's starting as FALSE, and only need to change when TRUE. But mostly I just think it's a matter of coder preference.

**Comment by DiegoPino** [comment]
I'm going to merge this so i can push your scheme storage changes and also the datastream label = filename fix. Thanks for testing @patdunlavey

**Pull request merged**